extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ConfigBeanCreator.java

```diff
@@ -27,7 +27,7 @@ public Object create(CreationalContext<Object> creationalContext, Map<String, Ob
             throw new IllegalStateException("Cannot load required type: " + requiredType);
         }
 
-        InjectionPoint injectionPoint = InjectionPointProvider.get();
+        InjectionPoint injectionPoint = (InjectionPoint) CreationalContextImpl.getExtra(creationalContext);
```

Comment:
I think that it would make sense to keep the `InjectionPointProvider.get()/set()` methods, add `CreationalContext` param and use `CreationalContextImpl.getExtra()` internally...

Reply:
This is a good point, I'm doing that as well.